packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'button_style.dart';
+import 'color_scheme.dart';
+import 'divider.dart';
+import 'icons.dart';
+import 'material.dart';
+import 'material_localizations.dart';
+import 'material_state.dart';
+import 'menu_theme.dart';
+import 'text_button.dart';
+import 'text_button_theme.dart';
+import 'text_theme.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+// Enable if you want verbose logging about menu changes.
+const bool _kDebugMenus = false;
+
+// How close to the edge of the safe area the menu will be placed.
+const double _kMenuViewPadding = 8.0;
```

Comment:
These values come from the spec - https://m3.material.io/components/menus/specs (layout section). Should they be derived from design tokens?